src/openvic-simulation/pop/Pop.cpp

```diff
@@ -97,27 +192,100 @@ bool PopManager::add_pop_type(
 		return false;
 	}
 	if (sprite <= 0) {
-		Logger::error("Invalid pop type sprite index for ", identifier, ": ", sprite);
+		Logger::error("Invalid pop type sprite index for ", identifier, ": ", sprite, " (must be positive)");
+		return false;
+	}
+	if (
+		(life_needs_income_types == NO_INCOME_TYPE) != (everyday_needs_income_types == NO_INCOME_TYPE) ||
+		(everyday_needs_income_types == NO_INCOME_TYPE) != (luxury_needs_income_types == NO_INCOME_TYPE)
+	) {
+		Logger::warning(
+			"Pop type ", identifier, " has inconsistent income types - all must be none, or all must be not none: life = ",
+			life_needs_income_types, ", everyday = ", everyday_needs_income_types, ", luxury = ", luxury_needs_income_types
+		);
+	}
+	if (max_size <= 0) {
+		Logger::error("Invalid pop type max size for ", identifier, ": ", max_size, " (must be positive)");
+		return false;
+	}
+	if (merge_max_size <= 0) {
+		Logger::error("Invalid pop type merge max size for ", identifier, ": ", merge_max_size, " (must be positive)");
+		return false;
+	}
+
+	if (research_points < 0) {
```

Comment:
Could be negative I think. Imagine a mod wants to simulate saboteur pops or something.

Reply:
Does that work in Vic2?